### PR TITLE
Source dependency on Microsoft.DotNet.SdkResolver

### PR DESCRIPTION
**Pull request by wli3**

### CODE REVIEW

**Review by joeloff — APPROVED** [review]

File: src/Resolvers/Microsoft.DotNet.MSBuildSdkResolver/Microsoft.DotNet.MSBuildSdkResolver.csproj [hunk 1]
```diff
@@ -33,8 +33,36 @@
...
   </ItemGroup>
 
+  <!-- To reduce dll load (cause RPS perf regression). Directly compile files from Microsoft.DotNet.SdkResolver -->
   <ItemGroup>
-    <ProjectReference Include="..\Microsoft.DotNet.SdkResolver\Microsoft.DotNet.SdkResolver.csproj" PrivateAssets="All" />
+    <Compile Include="..\Microsoft.DotNet.SdkResolver\**\*.cs" LinkBase="Microsoft.DotNet.SdkResolver" />
   </ItemGroup>
 
-</Project>
+  <ItemGroup Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'">
```
Comment (dsplaisted) [inline src/Resolvers/Microsoft.DotNet.MSBuildSdkResolver/Microsoft.DotNet.MSBuildSdkResolver.csproj:41]: This is all copied (and now duplicated from) the Microsoft.DotNet.SdkResolver project, right?

File: src/Tests/Microsoft.DotNet.MSBuildSdkResolver.Tests/Microsoft.DotNet.MSBuildSdkResolver.Tests.csproj [hunk 2]
```diff
@@ -13,7 +13,11 @@
   <ItemGroup>
     <ProjectReference Include="..\..\Microsoft.DotNet.TemplateLocator\Microsoft.DotNet.TemplateLocator.csproj" />
     <ProjectReference Include="..\..\Resolvers\Microsoft.DotNet.MSBuildSdkResolver\Microsoft.DotNet.MSBuildSdkResolver.csproj" />
-    <ProjectReference Include="..\..\Resolvers\Microsoft.DotNet.SdkResolver\Microsoft.DotNet.SdkResolver.csproj" />
+
+    <!-- Microsoft.DotNet.MSBuildSdkResolver compiles source of Microsoft.DotNet.SdkResolver to reduce dll load. There are duplicated types-->
+    <ProjectReference Include="..\..\Resolvers\Microsoft.DotNet.SdkResolver\Microsoft.DotNet.SdkResolver.csproj">
```
Comment (dsplaisted) [inline src/Tests/Microsoft.DotNet.MSBuildSdkResolver.Tests/Microsoft.DotNet.MSBuildSdkResolver.Tests.csproj:18]: Do we need to reference the DLLs from both types?  Should we just reference the MSBuildSdkResolver project and use the types from it?

**Review by dsplaisted — APPROVED** [review]

File: src/Resolvers/Microsoft.DotNet.MSBuildSdkResolver/Microsoft.DotNet.MSBuildSdkResolver.csproj (hunk 1 above)
Comment (wli3) [reply src/Resolvers/Microsoft.DotNet.MSBuildSdkResolver/Microsoft.DotNet.MSBuildSdkResolver.csproj:41]: yes

File: src/Tests/Microsoft.DotNet.MSBuildSdkResolver.Tests/Microsoft.DotNet.MSBuildSdkResolver.Tests.csproj (hunk 2 above)
Comment (wli3) [reply src/Tests/Microsoft.DotNet.MSBuildSdkResolver.Tests/Microsoft.DotNet.MSBuildSdkResolver.Tests.csproj:18]: I need to reuse the setup of testing msbuildsdkresolver to test template locator's finding VS version function. It is too much to duplicate, so i just added the template locator test in msbuildsdkresolver tests. And a type from template locator's dependency chain is used in the test. So i need Microsoft.DotNet.SdkResolver too

**Pull request merged**